### PR TITLE
Polaris JSON Read Writer, Unit Test, Materilization.

### PR DESCRIPTION
**Pull request by swapdewalkar**
--- PR TEMPLATE INSTRUCTIONS (1) ---
PR is regarding [issue](https://github.com/DAGWorks-Inc/hamilton/issues/418) 
Looking to submit a Hamilton Dataflow to the sf-hamilton-contrib module? If so go the the `Preview` tab and select the appropriate sub-template:
* [sf-hamilton-contrib template](?expand=1&template=HAMILTON_CONTRIB_PR_TEMPLATE.md)

Else, if not, please remove this block of text.

--- PR TEMPLATE INSTRUCTIONS (2) ---

Polars JSON read/write 

## Changes
Polaris JSON Read Writer, Unit Test, materializations

## How I tested this
unit test and materializations

## Checklist

- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future TODOs are captured in comments
- [x] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: hamilton/plugins/polars_extensions.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
+from __future__ import annotations
```
Comment (skrawcz) [inline hamilton/plugins/polars_extensions.py:1]: do we need to import this?

**Review by skrawcz — COMMENTED** [review]
Just a question on the import -- otherwise looking good thanks!

File: hamilton/plugins/polars_extensions.py (hunk 1 above)
Comment (swapdewalkar) [reply hamilton/plugins/polars_extensions.py]: Updated!

File: hamilton/plugins/polars_extensions.py [hunk 2]
```diff
@@ -434,6 +434,71 @@ def name(cls) -> str:
...
+
+
+@dataclasses.dataclass
+class PolarsJSONWriter(DataSaver):
+    """
+    Class specifically to handle saving JSON files with Polars.
+    Should map to https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.DataFrame.write_json.html
+    """
+
+    file: None = None
```
Comment (skrawcz) [inline hamilton/plugins/polars_extensions.py:477]:
```suggestion
    file: Union[IOBase, str, Path] 
```

Comment (skrawcz) [reply hamilton/plugins/polars_extensions.py:477]: having it None doesn't make sense with our materializer the way things are set up.

**Review by skrawcz — COMMENTED** [review]
this looks good. Will add my minor adjustment, and will merge. Thanks!

File: hamilton/plugins/polars_extensions.py [hunk 3]
```diff
@@ -434,6 +434,71 @@ def name(cls) -> str:
...
+
+@dataclasses.dataclass
+class PolarsJSONWriter(DataSaver):
+    """
+    Class specifically to handle saving JSON files with Polars.
+    Should map to https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.DataFrame.write_json.html
+    """
+
+    file: Union[IOBase, str, Path] 
+    pretty: bool = False
```
Comment (skrawcz) [inline hamilton/plugins/polars_extensions.py:478]:
```suggestion
    file: Union[IOBase, str, Path]
    pretty: bool = False
```

Comment (skrawcz) [reply hamilton/plugins/polars_extensions.py:478]: sorry trying to fix pre-commit.

**Review by skrawcz — APPROVED** [review]

**Pull request merged**